integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/outbox/OutboxPollingOutOfOrderIdsIT.java

```diff
@@ -120,13 +123,111 @@ public void processCreateUpdateDelete() {
 		backendMock.verifyExpectationsMet();
 	}
 
+	@Test
+	public void processDeleteRecreate_rightOrder() {
+		// An entity is deleted, then re-created in separate transactions.
+
+		int id = 1;
+		OrmUtils.withinTransaction( sessionFactory, session -> {
+			IndexedEntity entity = new IndexedEntity();
+			entity.setId( id );
+			entity.setIndexedField( "value for the field" );
+			session.persist( entity );
+		} );
+
+		backendMock.expectWorks( IndexedEntity.INDEX )
+				.add( "1", b -> b.field( "indexedField", "value for the field" ) )
+				.createdThenExecuted();
+		outboxEventFinder.showAllEventsUpToNow( sessionFactory );
+		backendMock.verifyExpectationsMet();
+
+		OrmUtils.withinTransaction( sessionFactory, session -> {
+			IndexedEntity entity = session.load( IndexedEntity.class, id );
+			session.remove( entity );
+		} );
+		OrmUtils.withinTransaction( sessionFactory, session -> {
+			IndexedEntity entity = new IndexedEntity();
+			entity.setId( id );
+			entity.setIndexedField( "value for the field" );
+			session.persist( entity );
+		} );
+
+		backendMock.expectWorks( IndexedEntity.INDEX )
+				.addOrUpdate( "1", b -> b.field( "indexedField", "value for the field" ) )
+				.createdThenExecuted();
+		outboxEventFinder.showAllEventsUpToNow( sessionFactory );
+		backendMock.verifyExpectationsMet();
+	}
+
+	@Test
+	@Ignore("Probably we need to create an issue for it")
```

Comment:
Ok, I see what the problem is. I created https://hibernate.atlassian.net/browse/HSEARCH-4228

Please mention the ticket here:

```suggestion
	@Ignore("HSEARCH-4228 Ensure correct processing of events even if their IDs are out of order")
```